src/wp-includes/media.php

```diff
@@ -1843,6 +1845,11 @@ function wp_filter_content_tags( $content, $context = null ) {
 				$filtered_image = wp_img_tag_add_loading_attr( $filtered_image, $context );
 			}
 
+			// Add 'decoding=async' attribute unless a 'decoding' attribute is already present.
```

Comment:
should we check for `attachment_id > 0` here similar to the checks above? cc: @felixarntz

Reply:
I don't think so. The attachment check is only done because `wp_img_tag_add_width_and_height_attr()` and `wp_img_tag_add_srcset_and_sizes_attr()` are passing the `$attachment_id` for those functions to look up the metadata. There is no metadata to look up for the `loading` attribute nor the `decoding` attribute, so in the same way that no `$attribute_id` check is done for `wp_img_tag_add_loading_attr()` there is likewise no need to check `wp_img_tag_add_decoding_async_attr()`. In fact, the `$attachment_id` should not be passed at all.

Reply:
👍🏼